add note in docs about DOCKER_HOST env var overriding unix socket config

camallen opened the pull request:
when trying to use docker without sudo and the DOCKER_HOST env var is set docker tries to connect to the specified host instead of using the unix socket.

GordonTheTurtle commented:
Can you please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "docker_host_docs" git@github.com:camallen/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~
This will update the existing PR, so you do not need to open a new one.

moxiegirl commented on docs/sources/installation/ubuntulinux.md:
```diff
@@ -169,6 +169,9 @@ To create the `docker` group and add your user:
 
 		$ docker run hello-world
 
+	If this fails with a message similar to below, check that the `DOCKER_HOST` env var is not set for your shell.
```
Thank you for the contribution.  Maybe go ahead and give them the recover. I'm just guessing it is this:

--> https://gist.github.com/moxiegirl/13a9870072a2ef4be15b

If this fails with a message similar this:

		Cannot connect to the Docker daemon. Is 'docker -d' running on this host?

Check that the `DOCKER_HOST` environ variable is not set for your shell. If it is, unset it.

camallen committed:
add note about DOCKER_HOST env var overriding unix socket config

when trying to use docker without sudo and the DOCKER_HOST env var is set docker tries to connect to the specified host instead of using the unix socket.

Signed-off-by: Campbell Allen <campbell.allen@gmail.com>

SvenDowideit commented:
mmm, if a user has had it set (on an ubuntu install?) thats a mildly advanced situation - but if that's a surprise, I'm going to guess someone else set it for them?

in which case, perhaps we should also tell them how to unset it?

jamtur01 commented:
I am not sure this error is always the result of that problem either.

camallen commented:
I agree that this specific env var config is most likely not the problem but I had a funky situation where my shell env was inherited from an old install and a fresh install of the OS + docker picked up the var. After much frustration I finally tweaked to the different behaviour between root and my user's shell env. I should have picked up on it quicker but in my defence I thought i had a clean install combined with unexpected behaviour (compared to the docs) and with a generic error message. 

I think a better error message reporting the specific host docker is failing to connect on would alleviate this issue. I had a quick look to add this but gave up and instead thought a note in the docs would help.

Unsetting the env var is most likely shell dependent, so specific instructions may lead to unmet expectations. How about I modify the docs to read something like: 
>If this fails with a message similar to this:
Cannot connect to the Docker daemon. Is 'docker -d' running on this host?

>Then it may be that the `DOCKER_HOST` environment variable is set for your
	shell and interfering with the default connection. Try to unset the environment
	variable in your shell, if this fixes the problem you should then remove it
	from your shell configuration.

LK4D4 commented:
@SvenDowideit What's up here?

SvenDowideit commented:
@LK4D4 I suspect this is one of many corner cases where the client can't find / talk to the daemon - and all of them would be better solved by adding code that tells the user more about what may be wrong in their situation.

idk, its not really ubuntu install specific, and its really hard to predict where different users will look to solve it.

crosbymichael commented:
@moxiegirl This one is 24days since last comment and a +6 line change.  Can you please take a look?

moxiegirl commented:
@crosbymichael Apologies, I was waiting for the other conversations to resolve.  I say let's merge it.  Might be a corner case...adding code might be another solution this will take care of it in the interim.

thaJeztah commented:
@camallen sorry, looks like this needs a rebase

camallen commented:
@thaJeztah done.

thaJeztah commented:
@SvenDowideit @jamtur01 @moxiegirl are we okay with merging this? I realize there are different causes to the same error, that cannot be described in a single case; this being one.


If we have no objection to that, then LGTM

thaJeztah commented:
ping @SvenDowideit @jamtur01

moxiegirl commented:
LGTM
